File: 2020_leetcode/problems/__02__/typescript/reverseString.ts
const reverseString = (str: string[]) => {
    let left = 0;
    let right = str.length - 1;
    while (left < right) {
        let tmp = str[left];
        str[left++] = str[right];
        str[right--] = tmp;
    }
    console.log(str);
}

reverseString('hello'.split(""))

File: 2020_leetcode/problems/__02__/kotlin/ReverseString.kt
fun reverseString(str: CharArray): Unit {
    var left = 0
    var right = str.size - 1
    while (left < right) {
        val tmp = str[left]
        str[left++] = str[right]
        str[right--] = tmp
    }
    println(str)
}
